### ISSUE TITLE
Changing Button Title/Description Text After Page Item is Displayed

### ISSUE THREAD

**Issue by edonv** [alexaubry/BulletinBoard#68]
### New Issue Checklist

- [x] I updated BulletinBoard to the latest version.
- [x] I read the [Contribution Guidelines](https://github.com/alexaubry/BulletinBoard/blob/master/CONTRIBUTING.md).
- [x] I read the [documentation](https://alexaubry.github.io/BulletinBoard).
- [x] I searched for [existing GitHub issues](https://github.com/alexaubry/BulletinBoard/issues).

### Issue Description

Something that's necessary for my use of BulletinBoard is the ability to change either button's title and description text after after page item has been displayed, as there is code being done in the background and I'm graying out the action button and changing its title. For some reason, when I try to set a button's title or the description text after it's been displayed, nothing happens. I'm not sure if this was intentional or not, but is there a workaround?

### Environment

- **iOS Version**: 11
- **Device(s)**: iOS Simulator

**Comment by eliburke** [comment]
_title_, _descriptionText_, and _image_ are all used to build the initial bulletin view, but do not provide direct access to the underlying UI components. You can modify them via PageBulletinItem.titleLabel.text, PageBulletinItem.descriptionLabel.text, and PageBulletinItem.imageView.text

Changing buttons is a little more involved. The "correct-est" way is to subclass PageBulletinItem and implement one of [headerViews, viewsUnderTitle, viewsUnderImage, viewsUnderDescription]. Inside, you can use the supplied BulletinInterfaceBuilder to call makeActionButton or makeAlternativeButton, save a reference in your class, then return it. Using the reference you can modify the button title as needed.

I don't have an easy example at hand, so this isn't tested code, but I believe it will work.

class LinkPreviewPage: PageBulletinItem {
    var callback: BulletinBoardCallback?
    var clickMeButton: UIButton!

    override func viewsUnderDescription(_ interfaceBuilder: BulletinInterfaceBuilder) -> [UIView]? {
        var contentViews = [UIView]()
        clickMeButton = interfaceBuilder. makeActionButton(title: "Click Me")
        contentViews.append(clickMeButton)
        return contentViews
    }
    
    override func tearDown() {
        super.tearDown()
        clickMeButton?.removeFromSuperview()
        clickMeButton = nil
    }
}

**Comment by edonv** [comment]
Thanks! The code you sent was almost perfect, but I figured it out. The makeActionButton() func returns a HighlightButtonWrapper instead of a UIButton, so the reference was of that type. And in the viewsUnderDescription override, I had to call add a target to the UIButton in the HighlightButtonWrapper referring to the actionButtonTapped func in the PageBulletinItem class. Once that was set up, I just need to call the setTitle func on the button reference.
